static/app/utils/createStorage.tsx

```diff
@@ -0,0 +1,31 @@
+// our storage wrapper is a subset of the full API
+type Storage = Pick<globalThis.Storage, 'getItem' | 'setItem' | 'removeItem'>;
+
+export default function createStorage(getStorage: () => globalThis.Storage): Storage {
+  try {
+    const storage = getStorage();
+    const mod = 'sentry';
+    storage.setItem(mod, mod);
+    storage.removeItem(mod);
+
+    return {
+      setItem: storage.setItem.bind(storage),
+      getItem: storage.getItem.bind(storage),
+      removeItem: storage.removeItem.bind(storage),
+    } as Storage;
+  } catch (e) {
+    return {
+      setItem() {
+        return;
+      },
+      // Returns null if key doesn't exist:
+      // https://developer.mozilla.org/en-US/docs/Web/API/Storage/getItem
+      getItem() {
+        return null;
+      },
+      removeItem() {
+        return null;
+      },
+    } as Storage;
```

Comment:
```
const storage: Storage = {...}
return storage
```
Alternatively you could define the noop storage outside of the createStorage fn, then just `return noopStorage`. If we avoid type casting, it will help us avoid some errors in the future if `Storage` interface changes because TS will warn us of the wrong signature (right now it wouldn't)

Reply:
@JonasBa good call, I just copy-pasted this code but no reason I can't make improvements :)

Reply:
Thanks so much for leaving it in better shape than you found it 🚀